### ISSUE TITLE
Speed-up CLIP model training with LightSeq

### ISSUE THREAD

**Issue by DenisKoposov** [bytedance/lightseq#309]
Hello! I am trying to replace layers in both CLIP transformers (text encoder and vit-b/32) with LSTransformerEncoderLayer. However, the model diverges in the middle of the first warmup epoch (LR is low, it actually behaves the same with any LR) in fp16 or stuck at the same loss value in fp32. I've also tried to replace only one transformer with LightSeq implementation, but the result is the same. The only setup that performs reasonable is the one with two LightSeq transformers with only 1 layer each. In this case the loss behaves as supposed to, however I didn't try to run the full training procedure for this.

Is there anything in LightSeq transformer implementation that can make it less stable or does not work at all in my case (contrastive image-text pretraining)?

I use torch 1.8.1 + cuda-10.2 + lightseq 2.21. My architecture is similar to the original provided by OpenAI (https://github.com/openai/CLIP).

**Update:**
I've managed to solve the problem myself. I found that during the model creation only the last layer bound weights and grads. The other layers didn't got their gradients. If you have a similar problem, check that you create a new config for each new layer. You can also use deepcopy for this purpose.
